components/brave_shields/core/browser/ad_block_component_filters_provider.cc

```diff
@@ -90,9 +91,17 @@ void AdBlockComponentFiltersProvider::UnregisterComponent() {
 
 void AdBlockComponentFiltersProvider::OnComponentReady(
     const base::FilePath& path) {
+  base::FilePath old_path = component_path_;
   component_path_ = path;
 
   NotifyObservers(engine_is_default_);
+
+  if (!old_path.empty()) {
```

Comment:
is the new adblock engine loaded at this point so nothing will read from the old one? I don't think posting a delayed task is a good way to deal deal with this if it's not

Reply:
At this point, the new path exists on disk and is used for future loads of the adblock engine. None of the on-disk contents of the directory are referenced by the engine after the initial load, just whatever's in-memory.

Reply:
I thought some files were read at runtime, but it sounds like that isn't the case so just make this BEST_EFFORT task priority

Reply:
updated